### ISSUE TITLE
[Skia] ms-appdata is not working properly

### ISSUE THREAD

**Issue by jeromelaban** [unoplatform/uno#8909]
### Current behavior

Loading an image using `ms-appdata` is not working properly.

### Expected behavior

Image is loaded properly from the local storage folder

### How to reproduce it (as minimally and precisely as possible)

Run this, given that `ApplicationData.Current.LocalFolder` contains the file to load.

```xaml
<Image Stretch="Uniform" MinHeight="200" MinWidth="200" Source="ms-appdata:///local/icon.png" />
```

### Workaround

Use a workaround like [the one mentioned here](https://github.com/unoplatform/uno/discussions/8905#discussion-4104057).

### Works on UWP/WinUI

Yes

### NuGet package version(s)

4.3.6

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

## Note for contributors

The following block is supposed to handle the loading properly:
https://github.com/unoplatform/uno/blob/574a441497e0e6faee8d389a3dbac1c40d4d7891/src/Uno.UI/UI/Xaml/Media/ImageSource.cs#L121-L125

But for some reason does not read the file at the right location.

**Comment by sarosigy** [comment]
In skia-gtk, on a linux, I also tried to use ms-appdata (with temp folder, download a bunch of images and display), with no success.
Setting BitmapImage UriSource to http is working, however.

Tried to find the cause, found these lines:
https://github.com/unoplatform/uno/blob/56d3ba98ee6cab055dc8d7c4941ce4494e20e946/src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.skia.cs#L35-L50

I got the idea, downloading to a folder below the application and set the UriSource to ms-appx:///blah is working.
It's kind of a workaround, but because it tries to find the scaled image, it has an overhead, unnecessary in this case.
